src/deepsparse/v2/text_generation/nl_engine_operator.py

```diff
@@ -86,43 +150,80 @@ def __init__(
 
         kwargs["engine_kwargs"] = engine_kwargs
         kwargs["model_path"] = onnx_file_path
+
         super().__init__(**kwargs)
 
-        self.input_ids_length = input_ids_length
+    def override_model(self, model_path: Union[str, Path], batch_size: int):
+        """
+        Override the model based on the provided batch_size, sequence_length,
+        and input_ids_length.
+        """
+        (
+            onnx_file_path,
+            output_indices_to_be_cached,
+            kv_cache_data_type,
+        ) = overwrite_onnx_model_inputs_for_kv_cache_models(
+            onnx_file_path=model_path,
+            batch_size=batch_size,
+            sequence_length=self.sequence_length,
+            input_ids_length=self.input_ids_length,
+        )
+        return onnx_file_path, output_indices_to_be_cached, kv_cache_data_type
 
-    def run(self, inp: NlEngineInput, **kwargs) -> Any:
+    def run(self, inp: NLEngineInputs, **kwargs) -> NLEngineOutputs:
         engine_input = inp.engine_inputs
         kv_cache = inp.kv_cache
 
-        inputs = self._add_kv_cache_to_input(engine_input, kv_cache)
-        if bool(kv_cache.engine_internal_cache):
+        split = True
+        if not isinstance(kv_cache, list):
+            split = False
+            kv_cache = [kv_cache]
+            engine_input = [engine_input]
+
+        inputs = list(map(self._add_kv_cache_to_input, engine_input, kv_cache))
+
+        if bool(kv_cache[0].engine_internal_cache):
             # conventionally, before dispatching
             # inputs to the engine, we validate them
             # if val_inp=True. However, in this case
             # we want to pass the empty kv cache inputs
             # (batch_size=0) to the engine. Therefore,
             # we skip the validation
+
+            # Internal kv_cache works for batch_size of 1 atm
```

Comment:
I'd rather leave explicit NotImplementedError than a comment

Reply:
well we dont use the continuous_batching_engine for this case. so the batch size will always be 1